x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/DoubleScriptFieldScript.java

```diff
@@ -35,55 +35,40 @@
         DoubleScriptFieldScript newInstance(LeafReaderContext ctx) throws IOException;
     }
 
-    private double[] values = new double[1];
-    private int count;
-
     public DoubleScriptFieldScript(Map<String, Object> params, SearchLookup searchLookup, LeafReaderContext ctx) {
         super(params, searchLookup, ctx);
     }
 
+    public abstract double[] execute();
+
     /**
      * Execute the script for the provided {@code docId}.
      */
-    public final void runForDoc(int docId) {
-        count = 0;
+    public final double[] runForDoc(int docId) {
         setDocument(docId);
-        execute();
+        return execute();
     }
 
-    /**
-     * Values from the last time {@link #runForDoc(int)} was called. This array
-     * is mutable and will change with the next call of {@link #runForDoc(int)}.
-     * It is also oversized and will contain garbage at all indices at and
-     * above {@link #count()}.
-     */
-    public final double[] values() {
-        return values;
+    public static double[] convertFromDouble(double v) {
```

Comment:
@stu-elastic and @jdconrad do these look right? I borrowed them from `FactoryTests`.

I see right now you force the converters to be static - would it be possible to make them non-static on the script?

Reply:
Not easily because the only "this" pointer we have is through class bindings.